plugins/woocommerce/client/legacy/js/admin/meta-boxes-product-variation.js

```diff
@@ -1459,6 +1500,12 @@ jQuery( function ( $ ) {
 					displaying_links.show();
 				}
 			}
+
+			// Show/hide variations empty state
+			setTimeout(
```

Comment:
The animation that removes the variation takes around 300ms. I added this timeout to prevent showing the empty state and the variation simultaneously (while it's being removed).

Reply:
I think the root cause of the problem is that this function, `set_paginav()` is called before the ajax call in `load_variations()` finishes (I put some `console.log()` statements in to trace things).

Calling `show_hide_variation_empty_state()` from within here doesn't really feel correct, as this function "Sets the pagenav fields".

It would be good to consider if this could either be called from somewhere else, or even inside `load_variations()` after the ajax succeeds. Or, perhaps trigger an event from within `load_variations()` that we could hook up to.